### ISSUE TITLE
Can we limit the scope of action?

### ISSUE THREAD

**Issue by dongfangyier** [Unity-Technologies/ml-agents#307]
hello everybody
       Can we limit the scope of action?
       I want to make a game of gobang, want to know how to keep the action falling in the free place all the time?

**Comment by vincentpierre** [comment]
Hi, without context, I would say you could try to penalize the agent for trying to do an impossible action and not realize the intended action. 
As of, now there is no way to dynamically change the action size if that is the question.

**Comment by dongfangyier** [comment]
Thank you for your reply.Can I modify the neural network by myself to achieve this effect? How do I generate binary files if I can?

**Comment by vincentpierre** [comment]
Hi,
There are some information [Here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Using-TensorFlow-Sharp-in-Unity-(Experimental).md) on how to generate the bytes file. You can also modify the model directly into `ppo/models.py`. The Neural Networks we use have a determined length for the output tensor (by construction). I still think modifying the `AgentStep` method is the simplest way to achieve your goal.

**Comment by dongfangyier** [comment]
Thank you for your prompt reply. I've tried to make this neural network play against traditional AI. I've also tried to get this Agent to train him with himself. But they didn't get the desired effect. Here is some of my code to ask you what advice you have for improving it.
![image](https://user-images.githubusercontent.com/20981311/35961421-9958613c-0ce8-11e8-98c6-ec5158c6a61b.png)

![image](https://user-images.githubusercontent.com/20981311/35961400-80f060e0-0ce8-11e8-9e34-bb77f8d131b7.png)
